pkg/virt-handler/vm.go

```diff
@@ -291,6 +292,12 @@ func (d *VMHandlerDispatch) processVmUpdate(vm *v1.VM, shouldDeleteVm bool) erro
 		return err
 	}
 
+	// Map Container Registry Disks to block devices Libvirt can consume
+	vm, err = registrydisk.MapRegistryDisks(vm)
+	if err != nil {
+		return err
+	}
```

Comment:
The other storage mechanisms that we integrate are different because they are the base layer "native" approaches to integration in k8s, where as this is adding a new mechanism which is  really a layer above the core storage backend infrastructure.

I'm don't seeing using REST as an item that blocks you - you could easily just implement a rest API in this patch series now. I would much rather see that generic approach taken straight away than to add this code that we know someone will have to rewrite later. Trying the REST approach straightaway, will also demonstrate the viability of that generic approach to dynamic storage integration